kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/commands/optaplanner/CreateSolverCommand.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.api.commands.optaplanner;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import org.kie.server.api.model.KieContainerResource;
+import org.kie.server.api.model.KieServerCommand;
+import org.kie.server.api.model.instance.SolverInstance;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+@XmlRootElement(name = "create-solver")
+@XStreamAlias( "create-solver" )
+@XmlAccessorType(XmlAccessType.NONE)
+public class CreateSolverCommand
+        implements KieServerCommand {
+
+    private static final long serialVersionUID = -1803374525440238478L;
+
+    @XmlElement
+    @XStreamAlias("solver-instance")
+    private SolverInstance instance;
+
+    public CreateSolverCommand() {
+        super();
+    }
+
+    public CreateSolverCommand(SolverInstance instance) {
+        this.instance = instance;
+    }
+
+    public SolverInstance getInstance() {
+        return instance;
+    }
+
+    public void setInstance(SolverInstance instance) {
+        this.instance = instance;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((instance == null) ? 0 : instance.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if ( this == obj )
+            return true;
+        if ( obj == null )
+            return false;
+        if ( getClass() != obj.getClass() )
+            return false;
+        CreateSolverCommand other = (CreateSolverCommand) obj;
+        if ( instance == null ) {
+            if ( other.instance != null )
+                return false;
+        } else if ( !instance.equals( other.instance ) )
```

Comment:
this will hardly ever be equal as SolverInstance doesn't implement equals (so they are only equal if they are the same instance, which is never).

I suggest removing the equals/hashcode code from this Command for now - until it's sure that we need to have such code.